src/Mvc/Mvc.ApiExplorer/test/EndpointMetadataApiDescriptionProviderTest.cs

```diff
@@ -456,7 +456,7 @@ static void AssertParameters(ApiDescription apiDescription)
                 apiDescription.ParameterDescriptions,
                 param =>
                 {
-                    Assert.Equal("Foo", param.Name);
+                    Assert.Equal("Foo", param.Name, ignoreCase: true);
```

Comment:
Since you added new tests to verify the correct parameter name casing, do you think is better to remove the line **488**. 

`AssertParameters(GetApiDescription(([AsParameters] ArgumentListRecordWithoutAttributes req) => { }, "/{foo}"));`

And do not `ignoreCase` here.